Assets/HoloToolkit/Input/Scripts/HandsManager.cs

```diff
@@ -21,45 +29,123 @@ public bool HandDetected
-        public delegate void HandInViewDelegate(bool handDetected);
-        public event HandInViewDelegate HandInView;
+        public bool HandsPressed
+        {
+            get { return pressedHands.Count > 0; }
+        }
+
+        /// <summary>
+        /// The world space position of the hand being used for the current manipulation gesture.
+        /// Valid only if a manipulation gesture is in progress.
+        /// </summary>
+        public Vector3 ManipulationHandPosition
+        {
+            get
+            {
+                Vector3 handPosition;
+                if (!currentHandState.properties.location.TryGetPosition(out handPosition))
+                {
+                    handPosition = Vector3.zero;
+                }
+                return handPosition;
+            }
+        }
+
+        private InteractionSourceState currentHandState;
 
         private HashSet<uint> trackedHands = new HashSet<uint>();
 
-        private void Awake()
+        private HashSet<uint> pressedHands = new HashSet<uint>();
+
+        private void OnEnable()
         {
             InteractionManager.SourceDetected += InteractionManager_SourceDetected;
+            InteractionManager.SourcePressed += InteractionManager_SourcePressed;
+            InteractionManager.SourceReleased += InteractionManager_SourceReleased;
+            InteractionManager.SourceUpdated += InteractionManager_SourceUpdated;
             InteractionManager.SourceLost += InteractionManager_SourceLost;
         }
 
+        private void OnDisable()
+        {
+            InteractionManager.SourcePressed -= InteractionManager_SourcePressed;
+            InteractionManager.SourceReleased -= InteractionManager_SourceReleased;
+            InteractionManager.SourceUpdated -= InteractionManager_SourceUpdated;
+            InteractionManager.SourceLost -= InteractionManager_SourceLost;
+        }
+
+        /// <summary>
+        /// Thrown when we detect a hand.
+        /// </summary>
+        /// <param name="state"></param>
         private void InteractionManager_SourceDetected(InteractionSourceState state)
         {
             // Check to see that the source is a hand.
-            if (state.source.kind != InteractionSourceKind.Hand)
+            if (state.source.kind == InteractionSourceKind.Hand)
             {
-                return;
+                trackedHands.Add(state.source.id);
+
+                if (HandInView != null)
+                {
+                    HandInView(HandDetected);
+                }
             }
+        }
+
+        /// <summary>
+        /// Thrown when the Interaction Source is pressed.
+        /// </summary>
+        /// <param name="state">The current state of the Interaction source.</param>
+        private void InteractionManager_SourcePressed(InteractionSourceState state)
+        {
+            if (state.source.kind == InteractionSourceKind.Hand)
+            {
+                if (!HandDetected)
```

Comment:
Yeah, this is strange.  I simply copied this over from the GestureManager.

I hate double negatives haha. I'll fix this too.